src/components/DatePickerRange.react.js

```diff
@@ -71,6 +71,16 @@ export default class DatePickerRange extends Component {
                 });
             }
         }
+
+        if (
+            clearable &&
+            !start_date &&
+            !end_date &&
+            oldMomentDates.start_date &&
+            oldMomentDates.end_date
+        ) {
+            setProps({start_date: null, end_date: null});
+        }
```

Comment:
Rather than preventing the update if one of start/end was not previously set, should instead check and add each individually if the condition is met.

e.g. this code is not optimal but gives the general idea
```
if (clearable && (!start_date || !end_date)) {
    const props = { };
    if (!start_date && oldMomentDates.start_date) {
        props.start_date = null;
    }
    // idem for end_date

    setProps(props);
}
```